src/chain/sync/grandpa_warp_sync.rs

```diff
@@ -52,12 +56,17 @@ pub struct Config {
     pub start_chain_information: ChainInformation,
     /// The initial capacity of the list of sources.
     pub sources_capacity: usize,
+    /// The chain spec of the chain.
+    pub chain_spec: ChainSpec,
```

Comment:
You can know the Babe configuration from the `ChainInformation`, no need to additionally pass the `ChainSpec`.